engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java

```diff
@@ -227,21 +227,13 @@ public void addPage(IPage page) {
 	}
 
 	protected void addPageRecord(IPage page, Connection conn) throws ApsSystemException {
-		int position = 1;
-		IPage[] sisters = page.getParent().getChildren();
-		if (null != sisters && sisters.length > 0) {
-			IPage last = sisters[sisters.length - 1];
-			if (null != last) {
-				position = last.getPosition() + 1;
-			} else {
-				position = sisters.length + 1;
-			}
-		}
+		String parentCode = page.getParent().getCode();
+		int position = this.getLastPosition(parentCode, conn) + 1;

```

Comment:
Add comment on the intent here. I think I get why its here but it is hiding some logic that will update the provided Object on what is effectively a save operation which can be really dangerous.

Reply:
I understand what you mean. In fact, if a new page is added the system assumes that it is always inserted in the last position (and any previous value of "position" is ignored), to avoid changes to the position of the "sister" pages.